src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -426,6 +432,26 @@ public String actualPathForPublicPath(String publicPath) {
         return null;
     }
 
+    /**
+     * Checks whether a path (defined by a URI) contains a certain sub-path (defined by a URI)).
+     *
+     * @param path    URI to be checked
+     * @param subPath URI to check
+     * @throws IllegalArgumentException if the provided path does not contain the provided sub-path
+     */
+    public static void sanitizeByCheckingIfPathContainsSubPathElseThrow(URI path, URI subPath) {
+        // Only act if path and subpath exist and if neither of them points to a temp location
+        if (path != null && !path.getPath().contains("files/temp") && subPath != null && !subPath.getPath().contains("files/temp")) {
+            // Removes redundant elements (e.g. ../ or ./) from the path and subPath
+            URI normalisedPath = path.normalize();
+            URI normalisedSubPath = subPath.normalize();
+            // Check whether the normalisedPath starts with the normalisedSubPath
+            if (!normalisedPath.getPath().startsWith(normalisedSubPath.getPath())) {
+                throw new IllegalArgumentException("Path is not valid!");
+            }
+        }
```

Comment:
I tried hard to find examples where this would miss some cases, but could not find any. Well done! ✅